src/DynamoCore/Interfaces/IWorkspaceModel.cs

```diff
@@ -0,0 +1,17 @@
+﻿using Dynamo.Models;
+using System;
+using System.Collections.Generic;
+
+namespace Dynamo.Interfaces
+{
+    public interface IWorkspaceModel
+    {
+        string Name { get; set; }
+
+        IEnumerable<ConnectorModel> Connectors { get; }
+        IEnumerable<NodeModel> Nodes { get; }
+
+        double CenterX { get; set; }
+        double CenterY { get; set; }
+    }
```

Comment:
Let's also add the `NodeAdded`, `NodeRemoved`, `NodesCleared`, `ConnectorAdded`, `ConnectorRemoved` events to this interface.